### PR TITLE
Feature/exit warning modal

### LINKED ISSUE #190: Warn professor when exiting a session before it is complete

### PR DESCRIPTION
**Pull request by weining-li**
Resolves #190

### CODE REVIEW

**Review by Goff-Davis — APPROVED** [review]
Code looks good to me but wait until LJ can approve because of my dynamo issues

**Review by kennygperez — DISMISSED** [review]

File: frontend/src/components/session-header/session-header.tsx
Comment (ljoks) [inline frontend/src/components/session-header/session-header.tsx:37]: This is an issue. you shouldn't be opening the modal and clearing the session. you should be clearing the session if they hit confirm on the modal. Meaning you probably have to copy the `clearSession` functionality to the modal component and call it there if they hit confirm

File: frontend/src/components/session-in-progress/session-progress/session-progress.tsx
Comment (ljoks) [inline frontend/src/components/session-in-progress/session-progress/session-progress.tsx:30]: I notice you're using `isClosed` of each sessionQuestion to determine if all questions have been responded to. I like this idea, but it's not always working. I think it's because of the incorrect dispatch i mentioned above ^

File: frontend/src/components/session-in-progress/session-controls/session-controls.tsx
Comment (ljoks) [inline frontend/src/components/session-in-progress/session-controls/session-controls.tsx:41]: you're dispatching this at the wrong time. this is getting dispatched whenever a professor hits "next question"

**Review by ljoks — CHANGES_REQUESTED** [review]
First over this has some bugs, which i mentioned some in the comments on your code. One extra one: 

- you also need to check for a modal popup when the professor hits `End Session` (this is in `session-controls.tsx`) Even if they are navigating to this button, it's possible that they skipped straight to the last question without presenting ones in the middle.

**Review by Goff-Davis — CHANGES_REQUESTED** [review]
You slipped an ngrok binary in `mobile/ngrok`. You'll need to remove that

File: frontend/src/components/session-in-progress/session-in-progress.tsx
Comment (Goff-Davis) [inline frontend/src/components/session-in-progress/session-in-progress.tsx:128]: There's no need to keep commented out html

**Review by Goff-Davis — APPROVED** [review]
Looks good, just needs lj to approve his change requests

**Review by ljoks — CHANGES_REQUESTED** [review]
Two quick fixes: 

1. When warning them why the session is incomplete, we should explain that the reason it's incomplete is all questions have not been 'Closed'. Otherwise it won't be clear what they need to do to not get the warning
2.  if you could just center the text in the modal i think it'd look a little better

Other than that lgtm

**Review by ljoks — APPROVED** [review]

**Pull request merged**
